class/Core/Uninstall.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Uninstall class
+ *
+ * @package notification
+ */
+
+namespace BracketSpace\Notification\Core;
+
+use BracketSpace\Notification\Utils\Files;
+use BracketSpace\Notification\Utils\View;
+use BracketSpace\Notification\Admin\Extensions;
+
+/**
+ * Uninstall class
+ */
+class Uninstall {
+
+	/**
+	 * Uninstall constructor
+	 */
+	public function __construct() {
+
+		register_uninstall_hook(NOTIFICATION_FILE, ['BracketSpace\Notification\Core\Uninstall', 'remove_plugin_data']);
+
+	}
+
+	/**
+	 * Removing plugin data
+	 *
+	 * @since  6.0.5
+	 * @return void
+	 */
+	public static function remove_plugin_data() {
+
+		global $wpdb;
+
+		$general_settings = get_option( 'notification_general' );
+
+		$un = $general_settings['uninstallation'];
+
+		// Remove notifications.
+		if ( isset( $un['notifications'] ) && 'true' === $un['notifications'] ) {
+			$wpdb->query( "DELETE FROM {$wpdb->posts} WHERE post_type = 'notification'" ); // phpcs:ignore
+		}
+
+		// Remove settings.
+		if ( isset( $un['settings'] ) && 'true' === $un['settings'] ) {
+
+			$settings_config = get_option( '_notification_settings_config' );
+
+			foreach ( $settings_config as $section_slug => $section ) {
+				delete_option( 'notification_' . $section_slug );
+				delete_site_option( 'notification_' . $section_slug );
+			}
+
+			delete_option( '_notification_settings_config' );
+			delete_option( '_notification_settings_hash' );
+
+		}
+
+		// Remove licenses.
+		if ( isset( $un['licenses'] ) && 'true' === $un['licenses'] ) {
+
+			$files            = new Files( '', '', '' );
+			$view             = new View( $files );
+			$extensions_class = new Extensions( $view );
```

Comment:
You don't need the `$files` nor `$view` instance, but more importantly, you don't even need to instantiate the `Extensions` class here. Please inject it to this class as it's already available in Runtime object under `admin_extensions` property.

If this is not instantiated during the uninstall process, you can leave it here, but no constructor params required.

Reply:
I did it the way it was in the previous uninstall process.

I don't have access to `admin_extensions`. This code creates a new instance of the class:
`register_uninstall_hook($this->plugin_file, [ 'BracketSpace\Notification\Core\Uninstall', 'remove_plugin_data' ]);`

I can't replace an instance stored in a variable in place of `'BracketSpace\Notification\Core\Uninstall'` because it causes an error. And that means I can't pass any parameters. Do you have any other idea?